src/renderer/modals/VoteTron/Body.js

```diff
@@ -0,0 +1,207 @@
+    footer: StepConfirmationFooter,
+  },
+];
+
+const mapStateToProps = createStructuredSelector({
+  device: getCurrentDevice,
+});
+
+const mapDispatchToProps = {
+  closeModal,
+  openModal,
+};
+
+const Body = ({
+  t,
+  stepId,
+  device,
+  closeModal,
+  openModal,
+  onChangeStepId,
+  params,
+  name,
+}: Props) => {
+  const [steps] = useState(createSteps);
+
+  const [optimisticOperation, setOptimisticOperation] = useState(null);
+  const [transactionError, setTransactionError] = useState(null);
+  const [signed, setSigned] = useState(false);
+
+  const {
+    transaction,
+    setTransaction,
+    account,
+    parentAccount,
+    status,
+    bridgeError,
+    bridgePending,
+  } = useBridgeTransaction(() => {
+    const { account } = params;
+
+    invariant(account && account.tronResources, "tron: account and tron resources required");
+
+    const { tronResources } = account;
+
+    const { votes } = tronResources;
+
+    const bridge = getAccountBridge(account, undefined);
+
+    const t = bridge.createTransaction(account);
+
+    const transaction = bridge.updateTransaction(t, {
+      mode: "vote",
+      votes,
+    });
+
+    return { account, parentAccount: undefined, transaction };
+  });
+
+  const handleCloseModal = useCallback(() => {
+    closeModal(name);
+  }, [closeModal, name]);
+
+  const handleStepChange = useCallback(e => onChangeStepId(e.id), [onChangeStepId]);
+
+  const handleRetry = useCallback(() => {
+    onChangeStepId("connectDevice");
+  }, [onChangeStepId]);
+
+  const handleTransactionError = useCallback((error: Error) => {
+    if (!(error instanceof UserRefusedOnDevice)) {
+      logger.critical(error);
+    }
+    setTransactionError(error);
+  }, []);
+
+  const handleOperationBroadcasted = useCallback(
+    (optimisticOperation: Operation) => {
+      if (!account) return;
+      updateAccountWithUpdater(account.id, account =>
+        addPendingOperation(account, optimisticOperation),
+      );
+      setOptimisticOperation(optimisticOperation);
+      setTransactionError(null);
+    },
+    [account],
+  );
+
+  const error = transactionError || bridgeError;
+
+  const stepperProps = {
+    title: t("vote.title"),
+    device,
+    account,
+    parentAccount,
+    transaction,
+    signed,
+    stepId,
+    steps,
+    errorSteps: [],
+    disabledSteps: [],
+    hideBreadcrumb: !!error,
+    onRetry: handleRetry,
+    onStepChange: handleStepChange,
+    onClose: handleCloseModal,
+    reward: params.reward,
+    error,
+    status,
+    optimisticOperation,
+    openModal,
+    setSigned,
+    onChangeTransaction: setTransaction,
+    onOperationBroadcasted: handleOperationBroadcasted,
+    onTransactionError: handleTransactionError,
+    t,
+    bridgePending,
+  };
+
+  return (
+    <Stepper {...stepperProps}>
+      <SyncSkipUnderPriority priority={100} />
+      <Track onUnmount event="CloseModalVote" />
+    </Stepper>
+  );
+};
+
+const C: React$ComponentType<OwnProps> = compose(
```

Comment:
Is there any reasons that this is not directly exported?

Reply:
I tried but i think its the typing on those compose components that are a bit tricky :thinking: